tcms/core/management/commands/refresh_permissions.py

```diff
@@ -0,0 +1,116 @@
+from django.apps import apps
+from django.conf import settings
+from django.contrib.auth.models import Group, Permission
+from django.core.management.base import BaseCommand
+
+
+class Command(BaseCommand):
+    help = ('Refresh permissions for Tester group '
+            '(set by DEFAULT_GROUPS setting) and remove stale ones.'
+            )
+
+    @staticmethod
+    def get_all_permissions():
+        """
+            Returns a set of all available permissions in format
+            app_label.perm_codename
+        """
+        permissions = list()
+        for perm in Permission.objects.all():
+            permissions.append("%s.%s" %
+                               (perm.content_type.app_label, perm.codename))
+        return set(permissions)
+
+    @staticmethod
+    def get_all_installed_apps():
+        """
+            Returns a set of currently installed applications as app_labels
+        """
+        applications = list()
+        for app_config in apps.get_app_configs():
+            applications.append(app_config.label)
+        return set(applications)
+
+    @staticmethod
+    def kiwi_apps():
+        """
+            Returns a list of Kiwi TCMS related applications with permissions
+            to be assigned to Tester group
+        """
+        kiwi_apps = ['bugs', 'django_comments', 'linkreference', 'management',
+                     'testcases', 'testplans', 'testruns', 'attachments']
+        return kiwi_apps
+
+    @staticmethod
+    def permissions_exclude():
+        """
+            Returns a set of permissions NOT to be assigned to Tester group
+        """
+        perm_exclude = {'attachments.delete_foreign_attachments', }
+        return perm_exclude
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--auto', '--noinput', '--no-input', action='store_true',
+            dest='noinput',
+            help='Automatic mode. Does not require user confirmation',
+        )
+
+        parser.add_argument(
+            '--quiet', action='store_true', dest='quiet',
+            help='Suppress most of the console output',
+        )
+
+    def assign_default_group_permissions(self, quiet=True, assign_if_none=False):
+        """
+            Assigns default permissions for Kiwi TCMS apps to Tester group
+            (set by DEFAULT_GROUPS setting)
+        """
+        if assign_if_none:
+            admin = Group.objects.get(name='Administrator')
+            if admin.permissions.count() == 0:
+                all_perms = Permission.objects.all()
+                admin.permissions.add(*all_perms)
+
+        tester_group = settings.DEFAULT_GROUPS[0]
+        tester = Group.objects.get(name=tester_group)
+        if not assign_if_none:
```

Comment:
this is `if not quiet`

Reply:
assign_if_none is used to implement the behavior of the old `assign_default_group_permissions()` (from `tcms/utils/permissions.py`. Now I see it's confusing and unnecessary. We don't need to check if Tester group has no permissions at all to avoid duplication conflicts. We check each permission instead. So we probably should skip adding all permissions to Administrators group when the function is called by `tcms/utils/permissions.py`? Or check and add them when the command is executed?

Reply:
It doesn't matter what `assign_if_none` does. You can't have 2 variables controlling the same type behavior and 1 variable controlling 2 different types of behavior. If you need the 2 adjust your function calls from the old places.